OpenKeychain/src/main/java/org/sufficientlysecure/keychain/ui/SettingsActivity.java

```diff
@@ -436,6 +436,18 @@ public void onResume() {
                     account,
                     ContactsContract.AUTHORITY
             );
+
+            SwitchPreference pref = (SwitchPreference) findPreference(Constants.Pref.ENABLE_WIFI_SYNC_ONLY);
```

Comment:
The entire `onPreferenceChangeListener` can be avoided now that you have set android:persistent=true.